packages/rs-drive-abci/src/abci/handler/mod.rs

```diff
@@ -381,14 +400,30 @@ where
         } else {
             let block_execution_outcome::v0::BlockExecutionOutcome {
                 app_hash,
-                tx_results,
+                state_transition_results,
                 validator_set_update,
             } = run_result.into_data().map_err(Error::Protocol)?;
 
+            // Get current platform version
+            let state = self.platform.state.read().expect("expected to get state");
```

Comment:
block platform state, not state.

Reply:
For prepare proposal as well?

Reply:
well yes, we want the version that the block was run on, not the version before we ran the block.

Reply:
That's actually version from the state. Please check the run block proposal function.